From 890a7157fe1228a105f68297b13fd5337a8ab233 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 22 Mar 2018 17:35:56 +0000
Subject: [PATCH 0830/1000] MANTIS 0049780 extension: ConveyorSBD: Send
 conveyor scanner devices platform.idl V3.7

added
  void logReaderOutput(in string unitID, in LoggingClass loggingClass, in ReaderOutputs readerData);



git-svn-id: svn://localhost/SelfServiceCommon/trunk@1330 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index 0c65cf4a..d1e71c37 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "3.6";
+    const string interface_version = "3.7";
 
     /** Data type for text lists. */
     typedef sequence<string> list;
@@ -254,6 +254,15 @@ module PlatformManagement
       string response;
       unsigned long long timestamp;        // utc time; number of milli seconds since 1.1.1970 00:00
     };
+    
+    // Scanner statistics - scanner devices, which have reported the data
+    typedef sequence<string> ReaderIds;
+    struct ReaderOutput
+    {
+        string          data;           // Scanned data
+        ReaderIds       readers;        // Originating devices
+    };
+    typedef sequence<ReaderOutput> ReaderOutputs;    
         
     enum ApplicationsState
     {
@@ -363,6 +372,8 @@ module PlatformManagement
         code setListener(in AppStateListener listener);
         
         code rmvListener(in AppStateListener listener);
+
+        void logReaderOutput(in string unitID, in LoggingClass loggingClass, in ReaderOutputs readerData);
 		};
 };
 
-- 
2.41.0.windows.1

